packages/components/dashboard/inventory-card/inventory-card.tsx

```diff
@@ -0,0 +1,289 @@
+                          title={t('Data source: Error')}
+                        />
+                      </div>
+                    )}
+                  {dataResourcesLoaded &&
+                    dataResourceStatusMap &&
+                    dataResourceStatusMap[PhaseType.PROCESSING] && (
+                      <div>
+                        <BlueInProgressIcon
+                          title={PhaseType.PROCESSING}
+                          className="icons"
+                        />
+                        <DataResourcesPopOver
+                          label={String(
+                            dataResourceStatusMap[PhaseType.PROCESSING]
+                          )}
+                          dataResources={dataResourceProcessingList}
+                          title={t('Data source: Processing')}
+                        />
+                      </div>
+                    )}
+                </>
+              ) : (
+                <Skeleton screenreaderText="Loading data resources" />
+              )}
+            </GalleryItem>
+            <GalleryItem className="inventory-card-item">
+              {obClaimsLoaded && !obClaimsError ? (
+                <>
+                  <div className="inventory-card-sub-item">
+                    <Link to={OBC_LIST_PATH}>
+                      {t('{{obcCount}} Object bucket claims', {
+                        obcCount: obClaims.length,
+                      })}
+                    </Link>
+                  </div>
+                  {obClaimsStatusMap && obClaimsStatusMap[PhaseType.ERROR] && (
+                    <div className="icons-container">
+                      <RedExclamationCircleIcon
+                        title={PhaseType.ERROR}
+                        className="icons"
+                      />
+                      <OBCPopOver
+                        label={String(obClaimsStatusMap[PhaseType.ERROR])}
+                        obcDetails={obcErrorList}
+                        title={t('Object bucket claim: Error')}
+                      />
+                    </div>
+                  )}
+                  {obClaimsStatusMap &&
+                    obClaimsStatusMap[PhaseType.PROCESSING] && (
+                      <div>
+                        <BlueInProgressIcon
+                          title={PhaseType.PROCESSING}
+                          className="icons"
+                        />
+                        <OBCPopOver
+                          label={String(
+                            obClaimsStatusMap[PhaseType.PROCESSING]
+                          )}
+                          obcDetails={obcProcessingList}
+                          title={t('Object bucket claim: Processing')}
+                        />
+                      </div>
+                    )}
+                </>
+              ) : (
+                <Skeleton screenreaderText="Loading object bucket claims" />
+              )}
+            </GalleryItem>
```

Comment:
make one generic component, that will take loaded, loadError, progressingList, errorList etc as a prop and returns required JSX.

Reply:
created one generic component and used it for all 

[commit](https://github.com/red-hat-storage/mcg-ms-console/pull/21/commits/53f4127634922b581be20aea0a73a32cb87c452e)

NOTE: Once you have seen all the changes, I will squash all the commits. Kept each major change separate so that it is easy to review and mention.